### PR TITLE
Port all relevant algos to use state.batch_set_item and state.batch_g…

### PR DESCRIPTION
**Pull request by eracah**
CO-24
Changed the following algorithms to use `state.batch_get_item` and `state.batch_set_item` instead of assuming the batch is a tuple and doing explicit unpacking and repacking:

* `colout`
* `cutmix`
* `cutout`
* `label_smoothing`
* `mixup`
* `progressive_resizing`
* `selective_backprop`

### CODE REVIEW

**Review by ravi-mosaicml — APPROVED** [review]
Overall LGTM, though see comments re: docstrings and type annotations. (Please make sure to adjust all files -- didn't comment them all)

In addition, before merging, please update the PR description to describe what's changed without referencing the JIRA issue.

File: composer/algorithms/colout/colout.py [hunk 1]
```diff
@@ -187,13 +187,21 @@ class ColOut(Algorithm):
...
+            is assumed to be first in the pair.
     """
 
     def __init__(self,
                  p_row: float = 0.15,
                  p_col: float = 0.15,
                  batch: bool = True,
-                 resize_target: Union[bool, str] = 'auto'):
+                 resize_target: Union[bool, str] = 'auto',
+                 input_key: Union[str, int, Callable, Any] = 0,
```
[the remark this answers is not in the document]
Comment (eracah) [reply composer/algorithms/colout/colout.py]: done in 
881d41c

File: composer/algorithms/colout/colout.py [hunk 2]
```diff
@@ -187,13 +187,21 @@ class ColOut(Algorithm):
         batch (bool, optional): Run ColOut at the batch level. Default: ``True``.
         resize_target (bool | str, optional): Whether to resize the target in addition to the input. If set to 'auto', resizing
             the target will be based on if the target has the same spatial dimensions as the input. Default: ``auto``.
+        input_key (str, int, or Callable): A key that indexes to the input
```
[the remark this answers is not in the document]
Comment (eracah) [reply composer/algorithms/colout/colout.py]: done in
881d41c

[the remark this answers is not in the document]
Comment (eracah) [reply composer/algorithms/colout/colout.py]: done in 
881d41c

**Pull request merged**